src/rollup/utils/persist/expiring-local-storage.js

```diff
@@ -0,0 +1,58 @@
+var PAYLOAD_KEY = '$';
+var EXPIRES_KEY = '@';
+
+function safeJsonParse (json) {
+  try {
+    return JSON.parse(json);
+  } catch (e) {
+    // recover
+  }
+}
+
+export default {
+  getItem: function (key) {
+    var textValue = localStorage.getItem(key);
+    var value = safeJsonParse(textValue);
+
+    if (value && EXPIRES_KEY in value) {
+      if (Date.parse(value[EXPIRES_KEY]) < Date.now()) {
+        localStorage.removeItem(key);
+        return null;
+      }
+      if (PAYLOAD_KEY in value) {
+        // Extend the expiration date:
+        this.setItem(key, value[PAYLOAD_KEY]);
+        return value[PAYLOAD_KEY];
+      }
+    }
+    return textValue;
+  },
+
+  setItem: function (key, payload, expiresOn) {
+    if (!expiresOn) {
+      expiresOn = new Date();
+      expiresOn.setDate(expiresOn.getDate() + 365);
+    }
+
+    var record = {};
+
+    record[PAYLOAD_KEY] = '' + payload;
+    record[EXPIRES_KEY] = expiresOn;
+
+    localStorage.setItem(key, JSON.stringify(record));
+  },
+
+  removeItem: function (key) {
+    localStorage.removeItem(key);
+  },
+
+  ttl: function (key, payload, milliseconds) {
```

Comment:
Where do we use this method?

Reply:
we don't currently use it. i left it here because it was already implemented (some of this code was harvested from webadmin), and I am planning on using it for implementing the planned config item (ttl: '1M' or whatever)... but let's remove it from this PR.

Reply:
we decided to keep it, since it's used in tests and we plan to use it when we parameterize expiration